conans/model/build_info.py

```diff
@@ -41,50 +47,134 @@ def __init__(self):
         self.description = None  # Description of the conan package
         # When package is editable, filter_empty=False, so empty dirs are maintained
         self.filter_empty = True
+        self._components = OrderedDict()
+
+    @property
+    def _sorted_components(self):
+        """
+        Sorted components from less dependent to the most one (less items in .deps attribute)
+        :return: ordered list of components
+        """
+        return sorted(self._components.values(), key=lambda component: len(component.deps))
+
+    @property
+    def libs(self):
+        if self._components:
+            result = []
+            for component in self._sorted_components:
+                for sys_dep in component.system_deps:
+                    if sys_dep and sys_dep not in result:
+                        result.append(sys_dep)
+                if component.lib:
+                    result.append(component.lib)
+            return result
+        else:
+            return self._libs
+
+    @libs.setter
+    def libs(self, libs):
+        if self._components:
+            raise ConanException("Setting first level libs is not supported when Components are "
+                                 "already in use")
+        self._libs = libs
+
+    @property
+    def exes(self):
+        if self._components:
+            return [component.exe for component in self._components.values() if component.exe]
```

Comment:
This should be sorted also probably.

Reply:
I don't' know what should be the criteria in this case